src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdEpoch.cpp

```diff
@@ -59,3 +62,21 @@ void JfrTraceIdEpoch::end_epoch_shift() {
   OrderAccess::storestore();
   _synchronizing = false;
 }
+
+bool JfrTraceIdEpoch::is_synchronizing() {
+  return Atomic::load_acquire(&_synchronizing);
+}
+
+void JfrTraceIdEpoch::set_method_tracer_tag_state() {
+  assert_locked_or_safepoint(ClassLoaderDataGraph_lock);
+  Atomic::store(&_method_tracer_state, true);
```

Comment:
Well, now that `has_method_tracer_changed_tag_state` does `Atomic::load_acquire`, the stores should be `Atomic::release_store` at very least? The fact these are done under the CLDG lock is not really enough.